src/Boot/src/ExceptionHandler.php

```diff
@@ -75,6 +75,10 @@ public static function handleShutdown(): void
      */
     public static function handleError($code, $message, $filename = '', $line = 0): void
     {
+        if (!(error_reporting() & $code)) {
```

Comment:
This is side effect. 

I think implementation like this should be prefer:
```php
public function __construct(int $reportingLevel = null)
{
    $this->reportingLevel = $reportingLevel ?? \error_reporting();
}

/// ...
if (!($this->reportingLevel & $code)) {
    return;
}
```
What do you think?

Reply:
I think side effect is expected here, though I agree generally it's not a good practice. `error_reporting()` operates as a global variable setter/getter, so if we cache it may confuse some future users. For example, if someone wants to turn off exceptions for a certain piece of code he should be able to do it by changing error_reporting twice.

I took a quick look at how other packages implement errorHandlers. Here is monolog:
```
    public function handleError($code, $message, $file = '', $line = 0, $context = [])
    {
        if ($this->handleOnlyReportedErrors && !(error_reporting() & $code)) {
            return;
        }
   ...
```

Here is sentry:
```
    private function handleError(int $level, string $message, string $file, int $line, ?array $errcontext = []): bool
    {
        if (0 === (error_reporting() & $level)) {
            $errorAsException = new SilencedErrorException(self::ERROR_LEVELS_DESCRIPTION[$level] . ': ' . $message, 0, $level, $file, $line);
        } else {
            $errorAsException = new \ErrorException(self::ERROR_LEVELS_DESCRIPTION[$level] . ': ' . $message, 0, $level, $file, $line);
        }
    ...
```